### ISSUE TITLE
"!!proc || !bin.finalPass" error in Bluetooth blocks

### ISSUE THREAD

**Issue by fredster33** [microsoft/pxt-microbit#4424]
**Describe the bug**
An error "!!proc || !bin.finalPass" occurs when some Bluetooth blocks are used.

**To Reproduce**
1. Go to 
https://makecode.microbit.org/_DCxarf9fm4x0
2. Click on "download"
3. See error

**Expected behavior**
There is no error, and the program compiles correctly.

**Screenshots**
![Screenshot](https://files.catbox.moe/b2pxtj.png)

**micro:bit version (please complete the following information):**
N/A (software)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Microsoft Edge Dev 
 - Version: 97

**Comment by fredster33** [comment]
This may or may not be related to #4421.

**Comment by martinwork** [comment]
I don't think this is a BLE problem as such. If the timeanddate calls and extension are removed, it compiles. 
It looks like it's the assert in https://github.com/microsoft/pxt/blob/master/pxtcompiler/emitter/emitter.ts#L2505.

**Comment by keble6** [comment]
Me too! No obvious reason. Sometimes disappears
